presto-hive/src/main/java/com/facebook/presto/hive/DirectoryLister.java

```diff
@@ -16,12 +16,12 @@
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.LocatedFileStatus;
 import org.apache.hadoop.fs.Path;
-import org.apache.hadoop.fs.RemoteIterator;
 
-import java.io.IOException;
+import java.util.Iterator;
+
+import static com.facebook.presto.hive.util.HiveFileIterator.NestedDirectoryPolicy;
 
 public interface DirectoryLister
 {
-    RemoteIterator<LocatedFileStatus> list(FileSystem fs, Path path)
```

Comment:
Curious: why do we change the return iterator from `RemoteIterator` to `Iterator`? I'm trying to understand the underlying assumption.

Reply:
`DirectoryLister` is always used in conjunction with `HiveFileIterator`, which is an `Iterator` and which manages the recursion for you.  This change is to push the recursion into the directory lister so you don't always need to create `HiveFileIterator`; therefore, it seemed appropriate (and easier from an implementation perspective) to make this return `Iterator`, since we always convert the output into `Iterator` anyway.